application/controllers/AdminController.php

```diff
@@ -120,6 +120,30 @@ public function error($message, $sURL = array())
         Yii::app()->end();
     }
 
+    public function actionRemoveTemplateToken()
+    {
+        if (isset(Yii::app()->session['templatetoken'])) {
+            unset(Yii::app()->session['templatetoken']);
+        }
+    }
+
+    public function actionInstallTemplateByToken()
```

Comment:
Why is this needed if we already installed template via CLI?

Reply:
This action is triggering the CLI command to install the template by token

Reply:
Also need a permission check here I think? Either super admin or extension access.

Reply:
If we wanna do template install via URL instead of message queue, I'd rather go via RPC API. Me and Carsten discussed it briefly, but decided to use the message queue system instead.

Reply:
Applied some changes to check for permissions